src/classes/GiftTemplate.cls

```diff
@@ -0,0 +1,67 @@
+public with sharing class GiftTemplate {
```

Comment:
It feels a bit strange to have both the GiftTemplate class and the GE_Template class. I guess GE_Template is pretty tightly coupled to the implementation and JSON structure we are storing it with, but for whatever reason this caught my attention. 

Did you end up resolving the issue with persisting permissions errors in the template JSON?

Reply:
It does to me as well, feels a bit incomplete. The primary reason I created this separately was so that whenever we work with form templates in Gift Entry specifically, we'd have a class that encompassed all the Gift Entry specific behavior related to form templates. Definitely not super tied to this particular implementation of that so I'd be happy to move this to GE_Template if we want.

And no, unfortunately, I haven't gotten around to resolving the issue with persisting perms error in the template json.